server/src/main/java/org/opensearch/index/query/IntervalsSourceProvider.java

```diff
@@ -631,6 +634,155 @@ String getUseField() {
         }
     }
 
+    public static class Regexp extends IntervalsSourceProvider {
+
+        public static final String NAME = "regexp";
+        public static final int DEFAULT_FLAGS_VALUE = RegexpFlag.ALL.value();
+
+        private final String pattern;
+        private final int flags;
+        private final String useField;
+        private final Integer maxExpansions;
+
+        public Regexp(String pattern, int flags, String useField, Integer maxExpansions) {
+            this.pattern = pattern;
+            this.flags = flags;
+            this.useField = useField;
+            this.maxExpansions = (maxExpansions != null && maxExpansions > 0) ? maxExpansions : null;
+        }
+
+        public Regexp(StreamInput in) throws IOException {
+            this.pattern = in.readString();
```

Comment:
Need version checks >= 1.3

Reply:
@mattweber do you think you need it? the 1.3.0+ would recognize `case "regexp"` but older clusters would not, so we should not see the case when older cluster tries to reconstruct `Regexp` from the stream

Reply:
Yea I am not sure, usually I do the checks for an existing feature not a new one.   Maybe we just go as is and see if we start triggering any BWC failures.  In which case I can do a followup PR for version checks.